### PR TITLE
Add recipe for PyTensor

### PR DESCRIPTION
**Pull request by michaelosthege**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by twiecki** [comment]
Acknowledged.

**Comment by maresb** [comment]
Willing to be comaintainer

**Comment by aseyboldt** [comment]
Willing to be comaintainer

File: recipes/pytensor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+      host:
+        - pip
+        - python
+      run:
+        - python
+
+        # Ref: <https://github.com/pymc-devs/pytensor/blob/rel-{{ version }}/setup.py>
+        - numpy>=1.17.0
+        - scipy>=0.14
```
Comment (ocefpaf) [inline recipes/pytensor/meta.yaml:34]: Sometimes this fails, not sure why it didn't here but better safe than sorry.

```suggestion
        - numpy >=1.17.0
        - scipy >=0.14
```

File: recipes/pytensor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+{% set version = "2.8.10" %}
+
+package:
+  name: pytensor-suite  # https://github.com/conda/conda-build/issues/3933
```
Comment (ocefpaf) [inline recipes/pytensor/meta.yaml:4]: Just a suggestion, not a requirement to merge this but I'd rather have comments on top and leave the inline `#` for selectors.

```suggestion
  # https://github.com/conda/conda-build/issues/3933
  name: pytensor-suite 
```

File: recipes/pytensor/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      commands:
+        - pytensor-cache help
+        - pip check
+        - python check-for-warnings.py allowed-warnings-base.txt
+      requires:
+        - pip
+      files:
+        - check-for-warnings.py
+        - allowed-warnings-base.txt
+  - name: pytensor
```
Comment (ocefpaf) [inline recipes/pytensor/meta.yaml:55]: Not important but a space here help us find sub-packages.

```suggestion

  - name: pytensor
```

File: recipes/pytensor/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+      host:
+        - pip
+        - python
+      run:
+        - python
+
```
Comment (ocefpaf) [inline recipes/pytensor/meta.yaml:32]:
```suggestion
```

**Comment by michaelosthege** [comment]
thank you, @ocefpaf !

I applied your suggestions and the pipeline is green again

**Pull request merged**